pkgs/applications/networking/browsers/ladybird/default.nix

```
@@ -1,71 +1,109 @@
 { lib
 , stdenv
 , fetchFromGitHub
+, fetchzip
+, cacert
+, tzdata
+, unicode-emoji
+, unicode-character-database
 , cmake
 , ninja
-, unzip
 , wrapQtAppsHook
 , libxcrypt
 , qtbase
+, qtmultimedia
 , nixosTests
 }:
 
+let
+  inherit (builtins) elemAt;
+  cldr_version = "43.1.0";
+  ucd_version = "15.0.0";
+  tzdb_version = "2023c";
+  cacert_version = "2023-05-30";
+  cldr-json = fetchzip {
+    url = "https://github.com/unicode-org/cldr-json/releases/download/${cldr_version}/cldr-${cldr_version}-json-modern.zip";
+    stripRoot = false;
+    hash = "sha256-2IxuzNFrww70JD6DruRjq5Kr7JrgtGOdhwaXLgRhZ8M=";
+    postFetch = ''
+      echo -n ${cldr_version} > $out/version.txt
+    '';
+  };
+in
 stdenv.mkDerivation (finalAttrs: {
   pname = "ladybird";
-  version = "unstable-2023-01-17";
+  version = "unstable-2023-08-12";
 
   src = fetchFromGitHub {
     owner = "SerenityOS";
     repo = "serenity";
-    rev = "45e85d20b64862df119f643f24e2d500c76c58f3";
-    hash = "sha256-n2mLg9wNfdMGsJuGj+ukjto9qYjGOIz4cZjgvMGQUrY=";
+    rev = "a0a1411ec96143e1d386bad7618898dc64669d4b";
+    hash = "sha256-Huz19OMTYHHrLMbILifDruG5xGTETopU40GViL4p1aM=";
   };
 
   sourceRoot = "${finalAttrs.src.name}/Ladybird";
 
   postPatch = ''
+    # Don't build macOS bundle, or refer to TARGET_BUNDLE_DIR
     substituteInPlace CMakeLists.txt \
-      --replace "MACOSX_BUNDLE TRUE" "MACOSX_BUNDLE FALSE"
-    # https://github.com/SerenityOS/serenity/issues/17062
-    substituteInPlace main.cpp \
-      --replace "./SQLServer/SQLServer" "$out/bin/SQLServer"
-    # https://github.com/SerenityOS/serenity/issues/10055
+      --replace "MACOSX_BUNDLE TRUE" "MACOSX_BUNDLE FALSE" \
+      --replace "if (APPLE)" "if (FALSE)"
+
+    # Don't set absolute paths in RPATH
     substituteInPlace ../Meta/Lagom/CMakeLists.txt \
-      --replace "@rpath" "$out/lib"
+      --replace "\''${CMAKE_INSTALL_BINDIR}" "bin" \
+      --replace "\''${CMAKE_INSTALL_LIBDIR}" "lib"
+  '';
+
+  preConfigure = ''
+    # Setup caches for LibLocale, LibUnicode, LibTimezone and LibTLS
+    mkdir -p build/Caches
+
+    ln -s ${cldr-json} build/Caches/CLDR
+
+    cp -r ${unicode-character-database}/share/unicode build/Caches/UCD
+    chmod +w build/Caches/UCD
+    cp ${unicode-emoji}/share/unicode/emoji/emoji-test.txt build/Caches/UCD
+    echo -n ${ucd_version} > build/Caches/UCD/version.txt
```

Can we get the versions from the packages? For example:

```suggestion
    echo -n ${unicode-emoji.version} > build/Caches/UCD/version.txt
```

---

If the version in the cache dir doesn't exactly match the version in our CMake download helpers, it will reject the fileb and try to download from the Internet. We would either need to make sure they always line up when updating the package, or lie to the Meta/CMake files with more patches/replacements.

---

I went ahead and did this for tzdata and unicode-character-database (the emoji-test.txt version is 15.0, not 15.0.0), and left some notes to future maintainers about the way the CMake files are particular about the exact versions matching.